utils/containers/lock_free/lf_queue.c.tpl

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "utils/containers/lock_free/lf_queue_{TYPE}.h"
+#include <stdlib.h>
+#include <string.h>
+
+void iota_lock_free_queue_umm_init_owner(iota_lf_umm_queue_{TYPE}_t* const queue,
+                                         uint32_t element_size) {
+  lfds711_queue_umm_init_valid_on_current_logical_core(
+      &queue->queue, &queue->queue_element_dummy, NULL);
+
+  lfds711_freelist_init_valid_on_current_logical_core(&queue->freelist, NULL, 0,
+                                                      NULL);
+}
+
+void iota_lock_free_queue_umm_init_user(iota_lf_umm_queue_{TYPE}_t* const queue) {
+  LFDS711_MISC_MAKE_VALID_ON_CURRENT_LOGICAL_CORE_INITS_COMPLETED_BEFORE_NOW_ON_ANY_OTHER_LOGICAL_CORE;
+}
+
+void* iota_lock_free_queue_umm_free(iota_lf_umm_queue_{TYPE}_t* const queue) {
+  if (queue == NULL) {
+    return;
+  }
+  lfds711_queue_umm_cleanup(&queue->queue, NULL);
```

Comment:
What happens to the elements not in the freelist but inside the queue? Shouldn't those be added to the freelist on cleanup iteration?